unix/x0vncserver/RandrGlue.c

```diff
@@ -0,0 +1,492 @@
+/* Copyright 2018 Peter Astrand for Cendio AB
+ * 
+ * This is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * 
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this software; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307,
+ * USA.
+ */
+
+#include <assert.h>
+#include <X11/Xlib.h>
+#ifdef HAVE_XRANDR
+#include <X11/extensions/Xrandr.h>
+#endif
+#include <stdlib.h>
+#include <string.h>
+
+#include "RandrGlue.h"
+
+#include <stdio.h>
+
+typedef struct _vncGlueContext {
+  Display *dpy;
+#ifdef HAVE_XRANDR
+  XRRScreenResources *res;
+#endif
+} vncGlueContext;
+
+static vncGlueContext _XDesktop_vncGlueContext;
+
+vncGlueContext *vncGetGlueContext()
+{
+  return &_XDesktop_vncGlueContext;
+}
+
+#ifdef HAVE_XRANDR
+static RRMode vncRandRGetPreferredMode(XRROutputInfo *output, XRRCrtcInfo *crtc,
+				       unsigned int width, unsigned int height)
```

Comment:
"Preferred" is a RANDR term, so perhaps we should avoid using that for other things.